src/Nethermind/Nethermind.Optimism/OptimismLegacyTxDecoder.cs

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Serialization.Rlp;
+using Nethermind.Serialization.Rlp.TxDecoders;
+using Nethermind.TxPool;
+
+namespace Nethermind.Optimism;
+
+public class OptimismLegacyTxDecoder : LegacyTxDecoder<Transaction>
+{
+    protected override Signature? DecodeSignature(ulong v, ReadOnlySpan<byte> rBytes, ReadOnlySpan<byte> sBytes, Signature? fallbackSignature = null,
+        RlpBehaviors rlpBehaviors = RlpBehaviors.None)
+    {
+        if (v == 0 && rBytes.IsEmpty && sBytes.IsEmpty)
+        {
+            return null;
+        }
+        return base.DecodeSignature(v, rBytes, sBytes, fallbackSignature, rlpBehaviors);
+    }
+}
+
+public class OptimismLegacyTxValidator(OptimismSpecHelper specHelper) : ITxValidator
+{
+    public ValidationResult IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec, BlockHeader? header = null)
+    {
+        if (header is null || specHelper.IsBedrock(header))
```

Comment:
Could IsBedrock be in `IReleaseSpec`, then we don't have to pass header?

`IReleaseSpec` was created to have all fork information.

Reply:
we can check if eip1559 is enabled. Not sure it's good idea. But as I know ReleaseSpec is not aware of optimism forks

Reply:
Would be good to make IReleaseSpec extendable by plugins then